avalon/houdini/lib.py

```diff
@@ -159,5 +160,23 @@ def unique_name(name, format="%03d", namespace="", prefix="", suffix="",
     return unique
 
 
+@contextlib.contextmanager
 def maintained_selection():
-    pass
+    """Maintain selection during context
+    Example:
+        >>> with maintained_selection():
+        ...     # Modify selection
+        ...     node.setSelected(on=False, clear_all_selected=True)
+        >>> # Selection restored
+    """
+
+    previous_selection = hou.selectedNodes()
+    try:
+        yield
+    finally:
+        if previous_selection:
+            for node in previous_selection:
+                node.setSelected(on=True)
+        else:
+            for node in previous_selection:
```

Comment:
This iteration is wrong since `previous_selection` at this point will always be empty. (since we are in the else statement).

This method will need a minor fix.